manifests/mountpoints/tmp.pp

```diff
@@ -126,16 +126,14 @@
             fstype   => 'none',
             options  => simplib::join_mount_opts(['bind'],$tmp_opts),
             device   => $facts['tmp_mount_path_tmp'],
-            remounts => true
+            remounts => true,
+            notify   => Exec['remount /tmp']
           }
 
-          if !empty(difference($tmp_opts,$_tmp_mount_tmp_opts)) {
```

Comment:
Unfortunately, this logic is correct. It's to prevent the case where the contents of `/etc/fstab` differ from the actual state of the mountpoint. This would be the case when someone maliciously modifies the running mountpoint options without updating fstab to match.

If there is an issue with flapping, then make sure that you have the latest `simplib` module in place. If that still does not work, you'll need to check the running system to see what the actual differences between the expected and running options are. You can insert a call to `simplib::debug::inspect` in this code to print out the different variables.